src/main/java/core/basesyntax/service/converter/DataConverterImpl.java

```
@@ -0,0 +1,32 @@
+package core.basesyntax.service.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String SPLIT = ",";
+    private static final int INDEX_TYPE = 0;
+    private static final int INDEX_FRUIT = 1;
+    private static final int INDEX_NUMBER = 2;
+
+    @Override
+    public List<FruitTransaction> convertAll(List<String> line) {
+        return line.stream()
+                .skip(1)
+                .map(this::lineSplit)
+                .map(f -> new FruitTransaction(
+                        Operation.getOperationType(f[INDEX_TYPE].toLowerCase()),
+                        f[INDEX_FRUIT], Integer.parseInt(f[INDEX_NUMBER])))
+                .collect(Collectors.toList());

```

General error-handling guidance: when a conversion error occurs (bad columns, invalid operation code, non-integer or negative quantity, empty fruit), rethrow a RuntimeException with an informative message and include the original exception where applicable (e.g., `new RuntimeException("Can't convert line: '" + row + "' : reason", e)`). This matches the checklist guidance: "Don't ignore exceptions in try-catch blocks ... rethrow a RuntimeException with an informative message and exception object."